plugins/inputs/http_listener_v2/http_listener_v2.go

```diff
@@ -268,6 +269,17 @@ func (h *HTTPListenerV2) collectBody(res http.ResponseWriter, req *http.Request)
 		return nil, false
 	}
 
+	// Handle snappy request bodies
+	if req.Header.Get("Content-Encoding") == "snappy" {
+		// Requires snappy decode for block format
+		bytes, err = snappy.Decode(nil, bytes)
```

Comment:
Let's use snappy.NewReader() instead which works on streams and not byte slices, then  ioutil.ReadAll is the last step of both cases, so definitely go with the switch statement

Reply:
As discussed refactored to use a switch statement, but sticking with Decode as the NewReader doesn't support the block format